src/main/java/io/confluent/connect/jdbc/TableQuerier.java

```diff
@@ -107,4 +121,11 @@ public int compareTo(TableQuerier other) {
       return this.name.compareTo(other.name);
     }
   }
+
+  private boolean exceptionIndicatesStalePreparedStatement(Connection db, SQLException e) throws SQLException {
+    // On PostgreSQL, queries with wildcards will be invalidated on schema changes
+    // and throw this error message.
+    return e.getMessage().contains("cached plan must not change result type") &&
```

Comment:
@gfodor This all seems reasonable. The thing I'm curious about is if we can do something better re: detecting the specific error that triggers this. First, is there a more specific exception we could catch that identifies the issue? And even if not, is there a better way to get the relevant info out of the exception that won't be sensitive to the specific formatting of the message? For example, https://github.com/rails/rails/commit/818d285305502cc6191a98400b43633f44394f6e seems to indicate there's a PG-specific error that (maybe?) we could detect.